pandas/core/dtypes/missing.py

```diff
@@ -124,61 +126,44 @@ def isna(obj):
 isnull = isna
 
 
-def _isna_new(obj):
-
-    if is_scalar(obj):
-        return libmissing.checknull(obj)
-    # hack (for now) because MI registers as ndarray
-    elif isinstance(obj, ABCMultiIndex):
-        raise NotImplementedError("isna is not defined for MultiIndex")
-    elif isinstance(obj, type):
-        return False
-    elif isinstance(obj, (ABCSeries, np.ndarray, ABCIndexClass, ABCExtensionArray)):
-        return _isna_ndarraylike(obj, old=False)
-    elif isinstance(obj, ABCDataFrame):
-        return obj.isna()
-    elif isinstance(obj, list):
-        return _isna_ndarraylike(np.asarray(obj, dtype=object), old=False)
-    elif hasattr(obj, "__array__"):
-        return _isna_ndarraylike(np.asarray(obj), old=False)
-    else:
-        return False
-
-
-def _isna_old(obj):
+def _isna(obj, old: bool = False):
```

Comment:
would it make sense to use, say, `inf_as_na` = True/False moving forward and remove the `old` concept for clarity. (follow-on ok too)

also could maybe use lib.no_default and look up options if not passed instead of the using the callback and partial.

Reply:
i agree this makes it more clear what is going on